### PR TITLE
Bug 1821778 - Add card for fullscreen research surface

### PR DESCRIPTION
**Pull request by sarah541**
| Portrait        | Landscape           |
| ------------- |:-------------:|
|      |  |

| Portrait        | Landscape           |
| ------------- |:-------------:|
|      |  |

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1821778

### CODE REVIEW

**Comment by mavduevskiy** [comment]
Just in case the [comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1821778#c3) on bugzilla might not get visibility, could we add design source to the bug?

**Review by rocketsroger — COMMENTED** [review]
The more I look at this the more it looks like [SimpleRedirectDialogFragment](https://github.com/mozilla-mobile/firefox-android/blob/main/android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/SimpleRedirectDialogFragment.kt).  I think if we're going this route we should add [newInstance](https://github.com/mozilla-mobile/firefox-android/blob/796b45fd9bbb81b54d9fc10f7133eb8a648cae16/android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/SimpleRedirectDialogFragment.kt#L60), and just use it similar to [here](https://github.com/mozilla-mobile/firefox-android/blob/796b45fd9bbb81b54d9fc10f7133eb8a648cae16/android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksFeature.kt#L135)

File: fenix/app/src/main/java/org/mozilla/fenix/experiments/ResearchSurfaceDialogFragment.kt [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+                putString(KEY_DISMISS_BUTTON_TEXT, keyDismissButtonText)
+            }
+
+            fragment.arguments = arguments
+            fragment.isCancelable = false
+
+            return fragment
+        }
+
```
Comment (rocketsroger) [inline fenix/app/src/main/java/org/mozilla/fenix/experiments/ResearchSurfaceDialogFragment.kt:109]: To call `showNow` I think we also need a tag here defined.  Something like:
        const val FRAGMENT_TAG = "MOZAC_RESEARCH_SURFACE_DIALOG_FRAGMENT"

File: fenix/app/src/main/res/navigation/nav_graph.xml [hunk 2]
```diff
@@ -33,6 +33,10 @@
         android:id="@+id/action_global_home_onboarding_dialog"
         app:destination="@id/homeOnboardingDialogFragment" />
 
+    <action
```
Comment (rocketsroger) [inline fenix/app/src/main/res/navigation/nav_graph.xml:36]: Don't need this anymore.

File: fenix/app/src/main/res/navigation/nav_graph.xml [hunk 3]
```diff
@@ -200,6 +204,23 @@
...
 
+    <dialog
+        android:id="@+id/researchSurfaceDialogFragment"
+        android:name="org.mozilla.fenix.experiments.ResearchSurfaceDialogFragment" >
+        <argument
+            android:name="KEY_MESSAGE_TEXT"
+            app:nullable="true"
+            app:argType="string" />
+        <argument
+            android:name="KEY_BUTTON_TEXT"
```
Comment (rocketsroger) [inline fenix/app/src/main/res/navigation/nav_graph.xml:215]: This should be "KEY_ACCEPT_BUTTON_TEXT"

File: fenix/app/src/main/res/navigation/nav_graph.xml [hunk 4]
```diff
@@ -200,6 +204,23 @@
...
+            app:nullable="true"
+            app:argType="string" />
+        <argument
+            android:name="KEY_URL"
```
Comment (rocketsroger) [inline fenix/app/src/main/res/navigation/nav_graph.xml:219]: This should be "KEY_DISMISS_BUTTON_TEXT"

File: fenix/app/src/main/res/navigation/nav_graph.xml [hunk 5]
```diff
@@ -200,6 +204,23 @@
         android:id="@+id/homeOnboardingDialogFragment"
         android:name="org.mozilla.fenix.onboarding.HomeOnboardingDialogFragment" />
 
+    <dialog
+        android:id="@+id/researchSurfaceDialogFragment"
+        android:name="org.mozilla.fenix.experiments.ResearchSurfaceDialogFragment" >
+        <argument
+            android:name="KEY_MESSAGE_TEXT"
+            app:nullable="true"
```
Comment (rocketsroger) [inline fenix/app/src/main/res/navigation/nav_graph.xml:212]: Don't need to be nullable.
This also applies to all of the arguments.

**Review by rocketsroger — COMMENTED** [review]
Almost there.  Some changes needed.

**Comment by rocketsroger** [comment]
After discussions with @Verdi, we've confirmed that this layout is good.  I'll have a follow up change to implement the correct fade in and slide up of the fragment.

**Pull request merged**